querybook/server/datasources/metastore.py

```diff
@@ -102,6 +102,23 @@ def get_table_by_name(
     return table_dict
 
 
+@register("/table_name/<schema_name>/<table_name>/metastore_link/", methods=["GET"])
+def get_table_metastore_link(
+    schema_name,
+    table_name,
+    metastore_id,
+    metadata_type,
+):
+    verify_metastore_permission(metastore_id)
+    metastore_loader = get_metastore_loader(metastore_id)
+
+    return metastore_loader.get_metastore_link(
```

Comment:
good point!

Reply:
I thought it over, perhaps we dont need to restrict the metastore link only for read-only mode . The metastore link exists no matter it's read-only or not, but as long as there is a link. For now, we only use it for redirection. but maybe we could still put the link some where even when it will write back to the metastore.